storage/src/index/storage.rs

```diff
@@ -0,0 +1,166 @@
+//! A memory efficient index structure for a journal that maps keys to locations.
```

Comment:
nit: Doesn't need a comment because `storage` isn't pub. If we keep, I'd reframe it to be about `storage.rs` rather than repeating `mod.rs`